src/azul/service/elasticsearch_service.py

```diff
@@ -329,39 +330,32 @@ def _generate_paging_dict(self, es_response, pagination, sort_facet):
         # ...else use search_after/search_before pagination
         es_hits = es_response['hits']['hits']
         count = len(es_hits)
-        next_ = {}
-        previous = {}
         if 'search_before' in pagination:
             # hits are reverse sorted
             if count > pagination['size']:
                 # There is an extra hit, indicating a previous page.
                 count = count - 1
                 search_before = es_hits[count - 1]['sort']
-                previous['search_before'], previous['search_before_uid'] = search_before
             else:
                 # No previous page
                 search_before = [None, None]
             search_after = es_hits[0]['sort']
-            next_['search_after'], next_['search_after_uid'] = search_after
         else:
             # hits are normal sorted
             if count > pagination['size']:
                 # There is an extra hit, indicating a next page.
                 count = count - 1
                 search_after = es_hits[count - 1]['sort']
-                next_['search_after'], next_['search_after_uid'] = search_after
             else:
                 # No next page
                 search_after = [None, None]
             if 'search_after' in pagination:
                 search_before = es_hits[0]['sort']
-                previous['search_before'], previous['search_before_uid'] = search_before
             else:
                 search_before = [None, None]
         sort = pagination.pop('sort')
         self_url = pagination.pop('self_url')
```

Comment:
If you remove the pops here, then you don't have to pass in `self_url` into page_link as a separate arg. You can just extract it from pagination. You don't have to pop sort either I think, you can just extract what you need when you compose the `page_field` dict.